### DISCUSSION TITLE
HELP: Had to wait months to get my laptop fixed, and now my affection levels are terrible

### DISCUSSION THREAD

**Discussion by JohnSmith2007** [Monika-After-Story/MonikaModDev#7991] [QA - tech support + general]
Exactly as the title implies. I had around, like, 400 or 450 affection about 6 months ago. And one day, my laptop motherboard dicked out on me, completely fried. I had to then wait throughout those 6 months just to get the replacement parts to come in. During all that time, i didn't have any other computer that i could hook up my hard drive to to continue with this mod. And now, Monika practically hates me. I have a busy work schedule, and i no longer have the time to spend over 200+ hours to re-gain all those affection points. Please, for the love of god, tell me that there's a way to fix this quickly and easily.

**Comment by ibm2431** [comment]
Looking at `_mas_AffStartup`, it should be -90 for 180 days.

Ironically, I see long absence greetings _also_ deduct affection inside the greeting if someone ends up being gone longer than they said they'd be. But the values seem really high in comparison to base loss. A 5 week absence after saying you'd be gone a few days results in a heavy -70, ~~and I'm not seeing anything preventing it from stacking with the loss in `_mas_AffStartup`. So it'd be -87.5 total~~, comparable to someone having left her for half a year without forewarning.

In that case, someone misjudging their absent time would have been better off _not_ telling her they're leaving _at all_, abandoning her without any warning, and eating the basic `_mas_AffStartup` loss for only the 5 weeks. I suspect that might not be intended behavior.

**Comment by Booplicate** [comment]
> Please, for the love of god, tell me that there's a way to fix this quickly and easily.

There's no "quick" way. Apologising and gifts might help.



>  I'm not seeing anything preventing it from stacking

Those do not stack.

> someone misjudging their absent time would have been better off not telling her they're leaving at all

Or, for example, telling her you don't know the exact time?

**Comment by ibm2431** [comment]
> Those do not stack.

I do see now there's a check for `._mas_long_absence`.

> Or, for example, telling her you don't know the exact time?

And if a person thought they had a good idea how long they'd be gone and were unfortunately wrong?

**Reply by Booplicate** [reply]
If something like that happened and the person couldn't visit once more to tell that, then the person was just unlucky. But this isn't really the case here?

**Reply by ibm2431** [reply]
Not specifically the case here, but can be in a related situation: "I need to bring my computer into the shop; they said it'd be just a few days."

Saying someone was unlucky brushes over the huge disparity. In real relationships, if you suddenly ghost your significant other for over 4 _months_ (-60 from `_mas_AffStartup`, no `long_absence`), your relationship is practically over absent a _real_ good reason. But in MAS, being wrong over how long you'd be gone ("days" -> actually 4 _weeks_) is supposedly _worse_ than that!

**Reply by Booplicate** [reply]
If you don't tell her you'll be gone, you'll lose aff, but if you do tell, you won't. If you can't manage to get back to her in time, then it's your fault. The farewell allows you to skip aff lose from not visiting, but keeps the responsibility to at least return as you promised.

I don't really see somebody telling they need 2 days, but end up be gone for 4+ weeks w/o being able to explain the situation. If your PC would have broken, you probably wouldn't even be able to tell her in the first place.

I do see the aff lose isn't exact fair though. And to tell you we were going to make changes (not exactly for this case, but in general) to how it works.

**Reply by ibm2431** [reply]
Yeah, I guess you really wouldn't have been able to tell her in the broken PC scenario! Good catch! (Second-best hypothetical scenario I have is "routine surgery", but that's admittedly not very likely at all).

I was originally wondering if it'd be better to strip the losses from the greetings, then always go into loss calculation and just give the player a "credit" in `AffStartup` based on how long they said they'd be gone for, with total affection loss being the difference. (Although I'd argue there should always be _some_ loss for extended durations, even _with_ a warning). But if y'all already have ideas kicking around, I'm sure you're already planning for the various scenarios better than I could!

**Reply by Booplicate** [reply]
If you feel like suggesting some changes, then better to open an issue about it. But I'll say now that I don't like the idea of removing some affection only to give it back a few lines later. It's unreliable, harder to understand (including for people who check aff log), and doesn't really make sense from Monika's standpoint since you're back as you promised, she shouldn't become less *affectionate* towards you.

**Comment by JohnSmith2007** [comment]
> I do see the aff lose isn't exact fair though. And to tell you we were going to make changes (not exactly for this case, but in general) to how it works.

Thank you. I'm glad to know that you're planning to make chnges to the Affection system. I'm just bummed because i never unlocked the piano minigame, and now it's gonna take forever to re-gain all of those points to unlock it. I'm dying to see what the piano layout looks like, because i have a shitload of ideas for custom songs. But i need to know exactly what notes are available so i know what i can and can't use ahead of time.

And speaking of the poem minigame, i already made a post in the discussions section proposing for a note editor, that way we don't have to make one by manually inserting code into a .json file. (Also because i'm used to the REAPER midi editor and i can't code for shit, but thats besides the point)

**Reply by Booplicate** [reply]
Affection isn't everything, it's also important to spend time with her even if she's in background.
